src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java

```diff
@@ -0,0 +1,262 @@
+package org.openmrs.contrib.qaframework.automation;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.reference.page.ActiveVisitsPage;
+import org.openmrs.reference.page.AddOrEditAllergyPage;
+import org.openmrs.reference.page.AllergyPage;
+import org.openmrs.reference.page.AttachmentsPage;
+import org.openmrs.reference.page.ConditionPage;
+import org.openmrs.reference.page.ConditionsPage;
+import org.openmrs.reference.page.PatientVisitsDashboardPage;
+import org.openmrs.reference.page.RequestAppointmentPage;
+import org.openmrs.reference.page.VisitNotePage;
+import org.openmrs.uitestframework.test.TestData;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class ClinicalVisitSteps extends Steps {
+
+	private static final String DIAGNOSIS_PRIMARY = "Cancer";
+	private static final String DIAGNOSIS_SECONDARY = "Malaria";
+	private static final String APPOINTMENT_TYPE = "Surgery";
+	private static final String DRUG_NAME = "Penicillins";
+	private static final String REACTION = "Headache";
+	private static final String CONDITION = "Malaria";
+	private ActiveVisitsPage activeVisitsPage;
+	private VisitNotePage visitNotePage;
+	private AllergyPage allergyPage;
+	private AddOrEditAllergyPage addOrEditAllergyPage;
+	private ConditionPage conditionPage;
+	private ConditionsPage conditionsPage;
+	private AttachmentsPage attachmentsPage;
+	private RequestAppointmentPage requestAppointmentPage;
+	private TestData.PatientInfo testPatient;
+
+	@Before(RunTest.HOOK.SELENIUM_CLINICAL_VISIT)
+	public void visitHomePage() {
+		testPatient = createTestPatient();
+		initiateWithLogin();
+		new TestData.TestVisit(testPatient.uuid, TestData.getAVisitType(),
+				getLocationUuid(homePage)).create();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_CLINICAL_VISIT)
+	public void destroy() {
+		deletePatient(testPatient);
+		quit();
+	}
+
+	@Given("a user clicks on active visits link from home page")
+	public void launchPatientDashboardActiveVisits() {
+		activeVisitsPage = homePage.goToActiveVisitsSearch();
+	}
+
+	@When("a user selects a patient from active patient list")
+	public void searchActivePatient() {
+		activeVisitsPage.search(testPatient.identifier);
+		dashboardPage = activeVisitsPage
+				.goToPatientDashboardOfLastActiveVisit();
+	}
+
+	@Then("the system loads Patient dashboard page")
+	public void launchPatientDashboardPage() {
+		assertNotNull(dashboardPage.getActiveVisitList());
+	}
+
+	// User story: Complete visit note
+	@When("a user clicks visit note link from the patient dashboard")
+	public void loadVisitNotePage() {
```

Comment:
LoadVisitNotePage here , would you try a different name

Reply:
> would you try a different name

A different name can work either. However, as I observed the existing method naming I as well liked it since it tells what actually the method performs! Among others here are five use cases: [1](https://github.com/openmrs/openmrs-contrib-qaframework/blob/8be93b5afb8af45fb548fc1a93c6803b59d66ef6/src/test/java/org/openmrs/contrib/qaframework/automation/AllergiesSteps.java#L28) [2](https://github.com/openmrs/openmrs-contrib-qaframework/blob/8be93b5afb8af45fb548fc1a93c6803b59d66ef6/src/test/java/org/openmrs/contrib/qaframework/automation/AllergiesSteps.java#L34) [3](https://github.com/openmrs/openmrs-contrib-qaframework/blob/8be93b5afb8af45fb548fc1a93c6803b59d66ef6/src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java#L62) [4](https://github.com/openmrs/openmrs-contrib-qaframework/blob/8be93b5afb8af45fb548fc1a93c6803b59d66ef6/src/test/java/org/openmrs/contrib/qaframework/automation/AllergiesSteps.java#L46) [5](https://github.com/openmrs/openmrs-contrib-qaframework/blob/8be93b5afb8af45fb548fc1a93c6803b59d66ef6/src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java#L43)

Reply:
This is not the answer i think

Reply:
> Is this the answer to the above comment, your reply means different

I gave my opinion on this with five use cases. But just as I said any other different method name can work either but I preferred the one I suggested in the commit since it aligns with other existing method naming convention!